### PR TITLE
[web] use semantic tags for headings (h1, h2, etc), and fix missing secondary roles

### PR DESCRIPTION
**Pull request by yjbanov**
Switch to using semantic heading tags (h1, h2, etc).

Fix missing secondary roles: focus, live region, route name, and label.
Improves indexability (https://github.com/flutter/flutter/issues/46789)

### CODE REVIEW

File: lib/web_ui/lib/src/engine/semantics/heading.dart [hunk 1]
```diff
@@ -3,35 +3,44 @@
...
-    setAriaRole('heading');
-  }
+  DomElement createElement() => createDomElement('h${semanticsObject.headingLevel}');
+
+  /// Focuses on this heading element if it turns out to be the default element
+  /// of a route.
+  ///
+  /// Normally, heading elements are not focusable as they do not receive
+  /// keyboard input. However, when a route is pushed (e.g. a dialog pops up),
+  /// the it may be desirable to move the screen reader focus to the heading
```
Comment (harryterkelsen) [inline lib/web_ui/lib/src/engine/semantics/heading.dart:28]: nit: the -> then

File: lib/web_ui/lib/src/engine/semantics/heading.dart [hunk 2]
```diff
@@ -3,35 +3,44 @@
...
+    addLabelAndValue(preferredRepresentation: LabelRepresentation.domText);
   }
 
   @override
-  bool focusAsRouteDefault() => focusable?.focusAsRouteDefault() ?? false;
-
-  void addHeadingRole() {
-    setAriaRole('heading');
-  }
+  DomElement createElement() => createDomElement('h${semanticsObject.headingLevel}');
```
Comment (harryterkelsen) [inline lib/web_ui/lib/src/engine/semantics/heading.dart:21]: What happens if the `headingLevel` changes? Does the update automatically create a new element and call this method?

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/semantics/heading.dart:21]: Heading level should not change. It confuses screen readers. Web crawlers snapshot the heading at its default value too, so changing has no effect either. We could issue a warning, but I'd rather such warnings are issued by the framework, and the engine will just do a graceful degradation of functionality (in this case the heading level stays the same forever).

File: lib/web_ui/lib/src/engine/semantics/heading.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/semantics/heading.dart:28]: Done.

**Review by harryterkelsen — APPROVED** [review]
LGTM

**Comment by cedvdb** [comment]
I thought one of the issue for doing this on the client was that Google indexation bot would not execute flutter and will just look at the plain index.html. that's not the case ?

**Pull request merged**